gnomad_qc/v5/sample_qc/create_sample_qc_metadata_ht.py

```diff
@@ -0,0 +1,634 @@
+    ht = ht.filter(ht.relationship != UNRELATED)
+
+    # Build a Table of relationships duplicating the info for each pair. Pair (i, j)
+    # will have two rows, one for (s: i.s, pair: j.s) and one for (s: j.s, pair: i.s).
+    ht = ht.select(*relationship_labels, s=ht.i.s, pair=ht.j.s).union(
+        ht.select(*relationship_labels, s=ht.j.s, pair=ht.i.s)
+    )
+
+    # Group the Table by the sample name (s) and aggregate to get a relationships
+    # dictionary per sample. The dictionary is built by grouping the relationship
+    # annotation (becomes the key) and collecting the set of all samples (value) with
+    # each relationship. Each item in filter_expr has its own annotation.
+    ht = ht.group_by(ht.s).aggregate(
+        **{
+            f"relationships{label}": hl.agg.filter(
+                hl.is_defined(ht[f"_relationship{label}"]),
+                hl.agg.group_by(
+                    ht[f"_relationship{label}"], hl.agg.collect_as_set(ht.pair)
+                ),
+            )
+            for label in filter_exprs
+        }
+    )
+
+    return ht
+
+
+def get_relationship_filter_expr(
+    hard_filtered_expr: hl.expr.BooleanExpression,
+    related_drop_expr: hl.expr.BooleanExpression,
+    relationship_set: hl.expr.SetExpression,
+    relationship: str,
+) -> hl.expr.builders.CaseBuilder:
+    """
+    Return case statement to populate relatedness filters in sample_filters struct.
+
+    :param hard_filtered_expr: Boolean for whether sample was hard filtered.
+    :param related_drop_expr: Boolean for whether sample was filtered due to
+        relatedness.
+    :param relationship_set: Set containing all possible relationship strings for
+        sample.
+    :param relationship: Relationship to check for. One of DUPLICATE_OR_TWINS,
+        PARENT_CHILD, SIBLINGS, or SECOND_DEGREE_RELATIVES.
+    :return: Case statement used to populate sample_filters related filter field.
+    """
+    return (
+        hl.case()
+        .when(hard_filtered_expr, hl.missing(hl.tbool))
+        .when(relationship == SECOND_DEGREE_RELATIVES, related_drop_expr)
+        .when(
+            hl.is_defined(relationship_set) & related_drop_expr,
+            relationship_set.contains(relationship),
+        )
+        .default(False)
+    )
+
+
+def annotate_relationships(ht: hl.Table, outlier_filter_ht: hl.Table) -> hl.Table:
+    """
+    Get relatedness relationship annotations for the combined meta Table.
+
+    Table has the following annotations:
+        - relationships: A dictionary of all relationships (except UNRELATED) the
+          sample has with other samples in the dataset. The key is the relationship
+          and the value is a set of all samples with that relationship to the given
+          sample.
+        - gnomad_v3_duplicate: Sample is in the gnomAD v3.1 sample set that passed hard
+          filtering.
+        - gnomad_v3_release_duplicate: Sample is in the gnomAD v3.1 release.
```

Comment:
Is this true? It doesnt look like these are added. We updated to gnomad and aou duplicates in the relatedness code. Should those be added? Or did you axe that for this more minimal representation?